src/io.rs

```diff
@@ -2,85 +2,105 @@
 
 use core::{fmt, slice};
 use core::fmt::Write;
-
-/// File descriptors
-const STDOUT: usize = 1;
-const STDERR: usize = 2;
+use nr;
 
 /// Host's standard error
-struct Stderr;
+pub struct Stderr {
+    fd: usize,
+}
+
+/// Construct a new handle to the host's standard error.
+pub fn stderr() -> Stderr {
```

Comment:
We can add a `try_stdout` function which returns a `Result`, and make `stdout` call it, so it looks more like `std::io::stdout`.